pages/account/[id].tsx

```diff
@@ -298,12 +294,48 @@ const Account = () => {
     [GraphQLSchema.AccountType.PolyjuiceContract].includes(accountType) ? { label: t('events'), key: 'events' } : null,
   ].filter(v => v)
 
+  const toChecksumAddress = (address: string) => {
+    address = address.toLowerCase().slice(2)
+    let hash = createKeccakHash('keccak256').update(address).digest('hex')
+    let ret = '0x'
+
+    for (let i = 0; i < address.length; i++) {
+      if (parseInt(hash[i], 16) >= 8) {
+        ret += address[i].toUpperCase()
+      } else {
+        ret += address[i]
+      }
+    }
+
+    return ret
+  }
+
+  const checkAddressIsValid = (address: string) => {
+    const len = address?.length || 0
+
+    if (len == 66) {
+      const sumResult = toChecksumAddress(address)
+      const uppercasedAddress = address.toUpperCase().slice(2)
+      return address.toLowerCase() === address || `0x${uppercasedAddress}` === address || address === sumResult
```

Comment:
Is this logic handling `godwoken script hash` instead of an eth address which consists of 20 bytes?

If yes, `godwoken script hash` is generated by `blake2b` as https://github.com/Keith-CY/gw-sdk/blob/master/packages/utils/src/index.ts#L7-L11, not `keccak256`, and it doesn't have a checksum.

Godwoken account/address works as a registry, it's a bit complex to explain in the comment, but `godwoken script hash` cannot be verified by an offline method.

Reply:
> Is this logic handling `godwoken script hash` instead of an eth address which consists of 20 bytes?
> 
> If yes, `godwoken script hash` is generated by `blake2b` as https://github.com/Keith-CY/gw-sdk/blob/master/packages/utils/src/index.ts#L7-L11, not `keccak256`, and it doesn't have a checksum.
> 
> Godwoken account/address works as a registry, it's a bit complex to explain in the comment, but `godwoken script hash` cannot be verified by an offline method.

Ok, I got it. I'll skip checking the `godwoken script hash`. @FrederLu